### PR TITLE
api enhancements 

### PR DESCRIPTION
**Pull request by KamiTzayig**
api update, includes all api needed for inviter, guard
and added server_defaults on models.py

### CODE REVIEW

File: src/routers/guard_router.py [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+from core.utilities import *
+from fastapi import Query
+
+
+
 
 router = APIRouter(prefix="/guard", tags=["guard"])
+
+
+def create_invitations_list():
```
Comment (inbarbarkai) [inline src/routers/guard_router.py]: Isn't it on utilities?

File: docker-compose.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ services:
     environment:
        IVT_OPENID_DISCOVERY_URL: 'http://keycloak:8080/auth/realms/master/.well-known/openid-configuration'
        IVT_OPENID_DISCOVERY_EXTERNAL_URL: 'http://keycloak.localhost/auth/realms/master/.well-known/openid-configuration'
-       IVT_ENABLE_AUTH: 'true'
+      #  IVT_ENABLE_AUTH: 'true'
```
Comment (inbarbarkai) [inline docker-compose.yml:17]: Are we sure we want this?

File: src/routers/guard_router.py [hunk 3]
```diff
@@ -1,3 +1,59 @@
...
+@router.get("/invitations", summary="Gets all the active invitations.")
+def read_invitations(invitations: InvitationRepository = Depends(create_invitations_list)):
+    today = datetime.today().strftime("%Y-%m-%d")
+    tomorrowRaw: datetime = datetime.today() + timedelta(days=2)
+    tomorrow = tomorrowRaw.strftime("%Y-%m-%d")
+    time_limit: datetime = (datetime.utcnow()-timedelta(hours=2)).strftime("%Y-%m-%dT%H:%M")
+
+    return invitations.query(lambda x: x.filter(or_(Invitation.is_active == True, Invitation.modify_timestamp >=time_limit), Invitation.invitees_arrival_timestamp >= today, Invitation.invitees_arrival_timestamp <= tomorrow))
+
+@router.get("/user-name", summary="Gets all the users.")
```
Comment (inbarbarkai) [inline src/routers/guard_router.py]: The name does not not describe the method functionality

File: src/routers/inviter_router.py [hunk 4]
```diff
@@ -1,12 +1,73 @@
...
+from fastapi import APIRouter, Depends, Path
 from database import InvitationRepository, create_connection, models, Invitation
+from datetime import datetime
+from pydantic import BaseModel
+from typing import Optional
 
 router = APIRouter(prefix="/inviter", tags=["inviter"])
 
+
 def create_invitations_list():
```
Comment (inbarbarkai) [inline src/routers/inviter_router.py]: Duplicate?

File: src/routers/inviter_router.py [hunk 5]
```diff
@@ -1,12 +1,73 @@
...
+def update_user(request: UpdateInvitationRequest, invitation: InvitationRepository = Depends(create_invitations_list),
+                invitation_id: str = Path(None, description="The unique identifier of the invitation.")):
+
+    update_data = request.dict(exclude_unset=True)
+    update_data['modify_timestamp'] = datetime.utcnow()
+    invitation.update_item(invitation_id, update_data)
+
+
+# @router.post("/edit-invitation/{invitation_id}", summary="Updates the invitation information.")
+# def update_user(request: UpdateInvitationRequest, invitation_id: str = Path(None, description="The unique identifier of the invitation."), invitation: InvitationRepository = Depends(create_invitations_list)):
```
Comment (inbarbarkai) [inline src/routers/inviter_router.py]: Remove commented code

**Review by inbarbarkai — CHANGES_REQUESTED** [review]
Align you variable and method names with pep8.

File: src/routers/guard_router.py [hunk 6]
```diff
@@ -1,3 +1,59 @@
...
+    # return {user.user_id: (user.first_name, user.last_name) for user in userList}
+    return dictList
+
+
+class UpdateInviteesRequest(BaseModel):
+    invitees_admitted: int
+
+
+@router.post("/change_admitted/{invitation_id}", summary="Updates the invitation information.")
+def update_user(request: UpdateInviteesRequest,  invitation: InvitationRepository = Depends(create_invitations_list),
```
Comment (inbarbarkai) [inline src/routers/guard_router.py]: The name does not not describe the method functionality

File: src/routers/inviter_router.py [hunk 7]
```diff
@@ -1,12 +1,73 @@
...
+    invitations.add_item(new_invitation)
+
+
+class UpdateInvitationRequest(BaseModel):
+    invitees_amount: Optional[int]
+    is_active: Optional[bool]
+    comment_for_guard: Optional[str]
+
+
+@router.post("/edit-invitation/{invitation_id}", summary="Updates the invitation information.")
```
Comment (inbarbarkai) [inline src/routers/inviter_router.py]: use underscore instead of hyphen

File: src/routers/guard_router.py [hunk 8]
```diff
@@ -1,3 +1,59 @@
...
+    today = datetime.today().strftime("%Y-%m-%d")
+    tomorrowRaw: datetime = datetime.today() + timedelta(days=2)
+    tomorrow = tomorrowRaw.strftime("%Y-%m-%d")
+    time_limit: datetime = (datetime.utcnow()-timedelta(hours=2)).strftime("%Y-%m-%dT%H:%M")
+
+    return invitations.query(lambda x: x.filter(or_(Invitation.is_active == True, Invitation.modify_timestamp >=time_limit), Invitation.invitees_arrival_timestamp >= today, Invitation.invitees_arrival_timestamp <= tomorrow))
+
+@router.get("/user-name", summary="Gets all the users.")
+def read_user_name(users: UsersRepository = Depends(create_users_repository), user_id_list: Optional[List[str]] = Query(None)):
+    userList:list(User) = users.query(lambda x: (x.filter(User.user_id.in_( user_id_list))))
```
Comment (inbarbarkai) [inline src/routers/guard_router.py]: The query itself should be implemented in `UserRepository`

File: src/routers/guard_router.py [hunk 9]
```diff
@@ -1,3 +1,59 @@
...
+
+
+@router.get("/invitations", summary="Gets all the active invitations.")
+def read_invitations(invitations: InvitationRepository = Depends(create_invitations_list)):
+    today = datetime.today().strftime("%Y-%m-%d")
+    tomorrowRaw: datetime = datetime.today() + timedelta(days=2)
+    tomorrow = tomorrowRaw.strftime("%Y-%m-%d")
+    time_limit: datetime = (datetime.utcnow()-timedelta(hours=2)).strftime("%Y-%m-%dT%H:%M")
+
+    return invitations.query(lambda x: x.filter(or_(Invitation.is_active == True, Invitation.modify_timestamp >=time_limit), Invitation.invitees_arrival_timestamp >= today, Invitation.invitees_arrival_timestamp <= tomorrow))
```
Comment (inbarbarkai) [inline src/routers/guard_router.py]: The query itself should be implemented in `InvitationRepository`

File: src/routers/inviter_router.py [hunk 10]
```diff
@@ -1,12 +1,73 @@
...
     return InvitationRepository(connection)
 
+
 @router.get("/invitations", summary="Gets all the invitations for this user.")
 def get_invitations(invitations: InvitationRepository = Depends(create_invitations_list), user_id: str = None):
-    return invitations.query(lambda x: x.filter(Invitation.user_id == user_id))
\ No newline at end of file
+    today = datetime.today().strftime("%Y-%m-%d")
+
+    return invitations.query(lambda x: x.filter(Invitation.user_id == user_id, Invitation.is_active == True, Invitation.invitees_arrival_timestamp >= today))
```
Comment (inbarbarkai) [inline src/routers/inviter_router.py]: The query itself should be implemented in `InvitationRepository`

File: docker-compose.yml (hunk 2 above)
Comment (KamiTzayig) [reply docker-compose.yml:17]: for the meantime the http doesn't work with keycloack as we don't have a way to authenticate aside from /docs

File: src/routers/inviter_router.py (hunk 4 above)
Comment (KamiTzayig) [reply src/routers/inviter_router.py]: we need the invitationRepository in guard and inviter, do you have a suggestion for a place to import "create_invitation_list" from?

Comment (inbarbarkai) [reply src/routers/inviter_router.py]: Yes, but I think it exist it utilities

File: docker-compose.yml (hunk 2 above)
Comment (inbarbarkai) [reply docker-compose.yml:17]: Ok

File: src/database/invitation_repository.py [hunk 11]
```diff
@@ -1,6 +1,29 @@
...
+from datetime import datetime, timedelta
+from sqlalchemy import DateTime, or_
+
+
 
 class InvitationRepository(RepositoryBase):
     def __init__(self, connection) -> None:
         super().__init__(connection, Invitation, Invitation.invitation_id)
+
+    def get_invitations_from_constrains(self):
```
Comment (inbarbarkai) [inline src/database/invitation_repository.py]: Constraints?

Comment (inbarbarkai) [reply src/database/invitation_repository.py]: Maybe pending?

Comment (KamiTzayig) [reply src/database/invitation_repository.py]: updated the function name

Comment (KamiTzayig) [reply src/database/invitation_repository.py]: get_pending_invitations

File: src/routers/inviter_router.py [hunk 12]
```diff
@@ -1,12 +1,60 @@
...
+
+
+class UpdateInvitationRequest(BaseModel):
+    invitees_amount: Optional[int]
+    is_active: Optional[bool]
+    comment_for_guard: Optional[str]
+
+
+@router.post("/edit_invitation/{invitation_id}", summary="Updates the invitation information.")
+def update_user(request: UpdateInvitationRequest, invitation: InvitationRepository = Depends(create_invitations_list),
```
Comment (inbarbarkai) [inline src/routers/inviter_router.py]: Still wrong method name

**Comment by inbarbarkai** [comment]
Great coding, now just fix all the pep8 violations and we're done

**Comment by yarons** [comment]
@KamiTzayig @inbarbarkai It can be done as a continuous effort: https://pypi.org/project/pep8/

**Comment by inbarbarkai** [comment]
@yarons from my experience, code written is not touched unless a fix needs to be introduced.
And even then, only the necessary parts.
Coding conventions are important for code readability.
I don't know a repo that will allow you to add work that doesn't follow it.

I want to say again that @KamiTzayig did really great work here.
He can take a little more time to for fixing those minor issues.

**Comment by yarons** [comment]
@inbarbarkai It's not for current code, it's for future code, think of it more like CI as part of the code, I can also create a GitHub action to check that while creating a PR.

**Comment by inbarbarkai** [comment]
Fine

**Review by inbarbarkai — APPROVED** [review]

**Pull request merged**